tests/kafkatest/utils/security_config.py

```diff
@@ -0,0 +1,132 @@
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import subprocess
+import tempfile
+
+
+class Keytool:
+
+    @staticmethod
+    def generate_keystore_truststore(dir='.'):
+        """
+        Generate JKS keystore and truststore and return
+        Kafka SSL properties with these stores.
+        """
+        ksPath = os.path.join(dir, 'test.keystore.jks')
+        ksPassword = 'test-ks-passwd'
+        keyPassword = 'test-key-passwd'
+        tsPath = os.path.join(dir, 'test.truststore.jks')
+        tsPassword = 'test-ts-passwd'
+        if os.path.exists(ksPath):
+            os.remove(ksPath)
+        if os.path.exists(tsPath):
+            os.remove(tsPath)
+        
+        Keytool.runcmd("keytool -genkeypair -alias test -keyalg RSA -keysize 2048 -keystore %s -storetype JKS -keypass %s -storepass %s -dname CN=systemtest" % (ksPath, keyPassword, ksPassword))
+        Keytool.runcmd("keytool -export -alias test -keystore %s -storepass %s -storetype JKS -rfc -file test.crt" % (ksPath, ksPassword))
+        Keytool.runcmd("keytool -import -alias test -file test.crt -keystore %s -storepass %s -storetype JKS -noprompt" % (tsPath, tsPassword))
+        os.remove('test.crt')
+
+        return {
+            'ssl.keystore.location' : ksPath,
+            'ssl.keystore.password' : ksPassword,
+            'ssl.key.password' : keyPassword,
+            'ssl.truststore.location' : tsPath,
+            'ssl.truststore.password' : tsPassword
+        }
+
+    @staticmethod
+    def runcmd(cmd):
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        proc.communicate()
+        if proc.returncode != 0:
+            raise subprocess.CalledProcessError(proc.returncode, cmd)
+
+
+class SecurityConfig:
+
+    PLAINTEXT = 'PLAINTEXT'
+    SSL = 'SSL'
+    SSL_DIR = "/mnt/ssl"
+    KEYSTORE_PATH = "/mnt/ssl/test.keystore.jks"
+    TRUSTSTORE_PATH = "/mnt/ssl/test.truststore.jks"
+
+    ssl_stores = Keytool.generate_keystore_truststore('.')
+
+    def __init__(self, node_account, security_protocol, template_props=""):
+        """
+        Initialize the security properties for the node and copy
+        keystore and truststore to the remote node if the transport protocol 
+        is SSL. If security_protocol is None, the protocol specified in the
+        template properties file is used. If no protocol is specified in the
+        template properties either, PLAINTEXT is used as default.
+        """
+
+        self.node_account = node_account
+        if security_protocol is None:
+            for line in template_props.split("\n"):
```

Comment:
Is the unparsed form of the properties file what we really want here? It seems like a simple `dict` would be easier here. Is this just because we get this data out of rendered templates? If nothing else, perhaps we should factor out the parsing of the config file into a utility?